### PR TITLE
refactor(tlsn-examples): update hyper and use http prover

### PR DESCRIPTION
**Pull request by sinui0**
This PR updates/simplifies the examples. It depends on both #433 and #432, otherwise the twitter example will fail.

# Changes
1. Bump version of `hyper` to `1.1` and update code to accommodate
2. Factor out some common code into the `tlsn-examples` lib
3. Update the twitter DM example to use `HttpProver` and new selective disclosure API.
4. Update twitter DM to use deferred decryption feature.

### CODE REVIEW

**Review by th4s — APPROVED** [review]
:+1:

**Review by themighty1 — APPROVED** [review]

File: tlsn/examples/twitter/twitter_dm.rs [hunk 1]
```diff
@@ -130,58 +112,29 @@ async fn main() {
...
-    });
-    // Commit to the received (public) data.
-    commitment_ids.push(builder.commit_recv(&(0..recv_len)).unwrap());
-
-    // Finalize, returning the notarized session
+    // Upgrade the prover to an HTTP prover, and start notarization.
+    let mut prover = prover.to_http().unwrap().start_notarize();
+
+    // Commit to the transcript with the default committer, which will commit using BLAKE3.
+    prover.commit().unwrap();
```
Comment (heeckhau) [inline tlsn/examples/twitter/twitter_dm.rs:127]: This fails for me:
```log
thread 'main' panicked at examples/twitter/twitter_dm.rs:127:21:
called `Result::unwrap()` on an `Err` value: HttpCommitError { idx: None, record_kind: Response, msg: "failed to commit to JSON body", source: Some(JsonCommitError { msg: "failed to commit array excluding values", source: Some(Duplicate(CommitmentId(122))) }) }
```

**Review by heeckhau — CHANGES_REQUESTED** [review]
Great work. The examples look a lot better ❤️ 

However, the Twitter examples fails for me now

**Comment by heeckhau** [comment]
> Rebased to include changes from #432 . @heeckhau merging is blocked on your review, please retry and confirm 🙏
@sinui0  I now get:
```
2024-02-12T23:03:53.833498Z DEBUG finalize: yamux::connection: 06e555ee: new outbound (Stream 06e555ee/63) of (Connection 06e555ee Client (streams 31))    
2024-02-12T23:03:54.614104Z DEBUG twitter_dm: Notarization complete!
thread 'main' panicked at examples/twitter/twitter_dm.rs:177:10:
called `Result::unwrap()` on an `Err` value: MissingCommitment
```

**Review by heeckhau — APPROVED** [review]
👍 Approved

**Pull request merged**